internal/service/vmservice/vm.go

```diff
@@ -384,6 +384,18 @@ func createVM(ctx context.Context, scope *scope.MachineScope) (proxmox.VMCloneRe
 	}
 
 	templateID := scope.ProxmoxMachine.GetTemplateID()
+	if templateID == -1 {
+		var err error
+
+		templateSelectorTags := scope.ProxmoxMachine.GetTemplateSelectorTags()
+		options.Node, templateID, err = scope.InfraCluster.ProxmoxClient.FindVMTemplateByTags(ctx, templateSelectorTags)
+
+		if err != nil {
```

Comment:
now the trick is here,

after you check `err != nil `
you need `if errors.Is(err, proxmox.ErrTemplateNotFound) {`

in this case you throw the Failure reason and message.

the normal case would be to retrun the error to the reconciler, so it can be requeed.

meaning, we only set Failure when the error is template not found,
other errors might occurs, e.g. network issue, proxmox api down, etc.

in that case, we need to return the error so the resource can try in the next reconciliation